src/compiler/checker.ts

```diff
@@ -51606,6 +51606,25 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
         let current: Node = node;
         while (current) {
             if (isFunctionLikeOrClassStaticBlockDeclaration(current)) {
+                if (node.label) {
+                    const functionOrClassLike = current as FunctionLikeDeclaration | ClassStaticBlockDeclaration;
+                    if (functionOrClassLike.body) {
+                        const relatedInfo = forEachChild(functionOrClassLike.body, childNode => {

```

Comment:
In an effort to try and finish the PR, I've restructured this to avoid the "incorrect" errors.